packages/react-components/react-components/src/AccessibilityScenarios/Textarea.stories.tsx

```diff
@@ -0,0 +1,350 @@
+import * as React from 'react';
+
+import { Button, Textarea, Checkbox, CheckboxOnChangeData, Label } from '@fluentui/react-components';
+
+import { Scenario } from './utils';
+
+import { useForm, Controller, OnSubmit } from 'react-hook-form';
+import { usePubSub, PubSubProvider, Handler } from '@cactuslab/usepubsub';
+
+const generateCustomerId = (): string => {
+  let char;
+  let hash = 0;
+  const random = Math.random().toString();
+  for (let i = 0; i < random.length; i++) {
+    char = random.charCodeAt(i);
+    //eslint-disable-next-line no-bitwise
+    hash = (hash << 5) - hash + char;
+    //eslint-disable-next-line no-bitwise
+    hash |= 0; // Convert to 32bit integer
+  }
+  hash += 2147483647; // Convert to positive integer
+  return hash.toString().padStart(10, '0').substring(2);
+};
+
+interface FormTextareas {
+  knowledge: string;
+  effort: string;
+  problemNotSolved: string;
+  otherComments: string;
+  satisfaction: string;
+}
+
+interface FormValidation {
+  subscribe: (channel: string, handler: Handler) => () => void;
+  unsubscribe: (channel: string, handler: Handler) => void;
+}
+
+interface ValidationMessageProps {
+  id: string;
+  formValidation: FormValidation;
+}
+const ValidationMessage: React.FC<ValidationMessageProps> = ({ id, formValidation, children }) => {
+  const [isAlerting, setIsAlerting] = React.useState(true);
+
+  const alert = React.useCallback(() => {
+    setIsAlerting(false);
+    setTimeout(() => setIsAlerting(true), 200);
+  }, [setIsAlerting]);
+
+  React.useEffect(() => {
+    formValidation.subscribe(id, alert);
+    return () => formValidation.unsubscribe(id, alert);
+  }, [formValidation, alert, id]);
+  return (
+    <div role={isAlerting ? 'alert' : undefined} style={{ color: isAlerting ? 'red' : 'green' }} id={`${id}Errors`}>
```

Comment:
I believe dynamically changing the role is going to be less robust than dynamically inserting the element into the DOM with a consistent `role="alert"`, and using a different element without an alert role for the non-alert message

Reply:
Okay, I have refactored it to insert a whole different DOM element instead of changing just role, though it looks like it works the same as before.